### DISCUSSION TITLE
Cant build: java.io.IOException: Cannot run program "rustc"

### DISCUSSION THREAD

**Discussion by enochc** [shadowsocks/shadowsocks-android#3175] [General]
When I try to build this I get a failure: java.io.IOException: Cannot run program "rustc"
Yes, I have rustc installed and cargo. I can run rustc from any location within the terminal. Why can't Gradle find it?

**Comment by AaronChen0** [comment]
Have you run
```
rustup target add armv7-linux-androideabi aarch64-linux-android i686-linux-android x86_64-linux-android
```

**Reply by enochc** [reply]
Yes I have, says they're all up to date. Thanks for the response.

**Reply by AaronChen0** [reply]
What system are you using to build?
What commands have you tried?
SDK and NDK version?
Also similar issue https://github.com/mozilla/rust-android-gradle/issues/126.
Do you modify this repo's source code to build?

**Comment by enochc** [comment]
Interestingly, I've discovered that I can build the APK just fine if I run the gradle command directly from the terminal: "./gradlew :mobile:assembleDebug" no problems, the error only shows up when I try to run the build from Android Studio, There is something with it's path that I just can't seem to figure out.
